components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -339,16 +351,56 @@ class AddonsManagerAdapter(
             }
         }
     }
+
+    /**
+     * Update the portion of the list that contains the provided [addon].
+     * @property addon The add-on to be updated.
+     */
+    fun updateAddon(addon: Addon) {
+        addonsMap[addon.id] = addon
+        submitList(createListWithSections(addonsMap.values.toList()))
+    }
+
+    /**
+     * Update only the portion of the list that changes between the current list and the new provided [addons].
+     * Be aware, updating a subset of the visible list is not supported, [addons] will replace
+     * the current list, but only the add-ons that have been changed will be updated in the UI.
+     * if you provide a subset it will replace the current list.
+     * @property addons A list of add-on to replace the actual list.
+     */
+    fun updateAddons(addons: List<Addon>) {
+        addonsMap = addons.associateBy({ it.id }, { it }).toMutableMap()
+        submitList(createListWithSections(addons))
+    }
+
+    private object DifferCallback : DiffUtil.ItemCallback<Any>() {
+        override fun areItemsTheSame(oldItem: Any, newItem: Any): Boolean {
+            return when {
+                oldItem is Addon && newItem is Addon -> oldItem.id == oldItem.id
```

Comment:
`oldItem.id == newItem.id`?

Reply:
Thanks 🎉 
It maybe the source of the issue that @csadilek encountered